File: Reactivities-API/Reactivities.Domain/Activities/Models/Activity.cs
using System.ComponentModel.DataAnnotations;
using Reactivities.Domain.Models;
using Reactivities.Domain.Users.Models;

namespace Reactivities.Domain.Activities.Models
{
    public class Activity
    {
        [Key]
        public long Id { get; set; }

        [Required]
        [StringLength(100)]
        public required string Title { get; set; }

        [Required]
        public DateTime Date { get; set; }

        [StringLength(400)]
        public string? Description { get; set; }

        [Required]
        [StringLength(20)]
        public required string Category { get; set; }

        [Required]
        [StringLength(100)]
        public required string City { get; set; }

        [Required]
        [StringLength(100)]
        public required string Venue { get; set; }

        [Required]
        public bool IsCancelled { get; set; }

        public required AppUser Host { get; set; }
        public ICollection<ActivityAttendee> Attendees { get; set; } = new List<ActivityAttendee>();
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    }
}

File: WebClient/src/app/models/activity.ts
import ActivityDto from './dtos/activityDto'
import User from './user'
import ProfileShort from './profileShort'

export interface Activity {
  id: number
  title: string
  date: Date
  description?: string
  category: string
  city: string
  venue: string
  isCancelled: boolean
  host: ProfileShort
  attendees: ProfileShort[]

  isHost: boolean
  isGoing: boolean
}

export class Activity implements Activity {
  constructor(init: ActivityDto, user: User | null) {
    Object.assign(this, init)

    if (user) {
      this.isGoing = this.attendees!.some((a) => a.username === user.username)
      this.isHost = this.host.username === user.username
    }

    this.date = new Date(this.date!)
  }
}
